### PR TITLE
feat(portals): addressing load times

### PR DESCRIPTION
**Pull request by eric-horodyski**

### CODE REVIEW

File: docs/portals/knowledge-base/addressing-portal-load-times.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+title: Addressing Portal Load Times
+---
+
+import Admonition from '@theme/Admonition';
+
+As you work with Portals you may notice a brief period where the Portal appears blank, which is not an ideal user experience. The duration of this period varies based on device performance and the size of the web assets being loaded within the Portal. 
+
+To effectively address this issue, it's important to understand why it occurs. Portals builds atop the system WebView provided by iOS and Android, with the system WebView responsible for initiating the loading of web assets. Portal components adhere to the loading lifecycle of the system WebView: they begin loading once added to the view stack and stop running when removed from the view stack. As a result, it's natural to experience an initial "blank" period when adding a Portal component to the view stack. 
+
+To minimize the impact web assets loading causes to the user experience:
```
Comment (jameson-w-taylor) [inline docs/portals/knowledge-base/addressing-portal-load-times.md:11]: maybe add a "you can" to the end here, so it sounds more natural?
> ...to the user experience, you can:

File: docs/portals/knowledge-base/addressing-portal-load-times.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+## Mask with a loading view
+
+One effective way to prevent users from seeing "blank Portals" is to add the Portal component to the view stack with no opacity to hide it as it loads, and display a loading view until the web app is ready to be presented. Once the web app is ready to be viewed, remove the loading view and set the Portal component's opacity to 1 to display it with a loaded web app.
```
Comment (jameson-w-taylor) [inline docs/portals/knowledge-base/addressing-portal-load-times.md:15]: "web app" seems to be repeated an awful lot here, and while it's important to be specific about which component you're talking about I think this can be simplified a little.

>...until the web app is ready to be presented. Once it is ready, remove the loading view and set the Portal component's opacity to 1 to display the rendered content.

File: docs/portals/knowledge-base/addressing-portal-load-times.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+There are two ways to approach signaling when the Portal is ready to be made visible. You can use the [communication mechanisms](https://ionic.io/docs/portals/choosing-a-communication) built into Portals to let the web app notify the native code layer when it is ready for the Portal to be visible. Tutorials on implementing this approach can be found here [here for iOS](https://ionic.io/docs/portals/for-ios/how-to/define-api-in-typescript) and [here for Android](https://ionic.io/docs/portals/for-android/how-to/define-api-in-typescript). Alternatively, you can use Portals' [Web Vitals plugin](https://ionic.io/docs/portals/for-web/web-vitals) to supply a callback that will display the Portal after First Contentful Paint (FCP) has completed.
```
Comment (jameson-w-taylor) [inline docs/portals/knowledge-base/addressing-portal-load-times.md:17]: there's an extra "here" in there:

>... can be found here here for iOS and ...

File: docs/portals/knowledge-base/addressing-portal-load-times.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+## Optimize the web app's load time
+
+Implementing a loading view will prevent the user from seeing "blank Portals", but it does not impact the duration it takes your web app to load. You can use Portals' [Web Vitals plugin](https://ionic.io/docs/portals/for-web/web-vitals) to measure the web app's performance.
```
Comment (jameson-w-taylor) [inline docs/portals/knowledge-base/addressing-portal-load-times.md:21]: Maybe reword this to suggest that the "loading view" approach shouldn't be considered a replacement for optimizing the web app performance.

>... but this does nothing to fix the scenario of when a poorly optimized web app may be slow to load. While it does provide a better user experience, you should also be taking steps to manage your startup performance. You can use ...

**Review by jameson-w-taylor — CHANGES_REQUESTED** [review]

File: docs/portals/knowledge-base/addressing-portal-load-times.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+Focus on reducing the [First Contentful Paint](https://web.dev/articles/fcp) to ensure the web app starts rendering quickly. This may mean the introduction of skeleton screens in the web app while data is being fetched from the network, etc. You can learn more about profiling your web application [here for iOS](https://ionic.io/docs/portals/for-web/ios-profiling) and [here for Android](https://ionic.io/docs/portals/for-web/android-profiling).
```
Comment (mattrein) [inline docs/portals/knowledge-base/addressing-portal-load-times.md:23]: Consider putting this earlier in the doc and make it clear that it should be part of either solution they choose.

**Review by mattrein — APPROVED** [review]

**Review by jameson-w-taylor — APPROVED** [review]

File: docs/portals/knowledge-base/addressing-portal-load-times.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+title: Addressing Portal Load Times
+---
+
+import Admonition from '@theme/Admonition';
+
+As you work with Portals you may notice a brief period where the Portal appears blank, which is not an ideal user experience. The duration of this period varies based on device performance and the size of the web assets being loaded within the Portal. 
+
+To effectively address this issue, it's important to understand why it occurs. Portals builds atop the system WebView provided by iOS and Android, with the system WebView responsible for initiating the loading of web assets. Portal components adhere to the loading lifecycle of the system WebView: they begin loading once added to the view stack and stop running when removed from the view stack. As a result, it's natural to experience an initial "blank" period when adding a Portal component to the view stack. 
+
+There are some ways you can minimize the impact loading web assets causes to the user experience:
```
Comment (kensodemann) [inline docs/portals/knowledge-base/addressing-portal-load-times.md:11]: This just sort of leaves the user hanging. I think it would be better to tell them what you will explain below, then explain it below. For example:

"To minimize the impact on the user experience, you can optimize the app's load time and mask the loading of the Portal component with a loading view."

Or

There are some ways you can minimize the impact loading web assets causes to the user experience:

- Optimize the web app's load time
- Mask the Portal with a loading view

File: docs/portals/knowledge-base/addressing-portal-load-times.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+## Mask with a loading view
+
+Use a loading view to prevent users from seeing "blank Portals". Add the Portal component to the view stack with zero opacity to hide it as it loads and display the loading view until the web app is ready to be presented. Once ready, remove the loading view and set the Portal component's opacity to 1 to display the rendered content.
+
+There are two ways to approach signaling when the Portal is ready to be made visible. You can use the [communication mechanisms](https://ionic.io/docs/portals/choosing-a-communication) built into Portals to let the web app notify the native code layer when it is ready for the Portal to be visible. Tutorials on implementing this approach can be found [here for iOS](https://ionic.io/docs/portals/for-ios/how-to/define-api-in-typescript) and [here for Android](https://ionic.io/docs/portals/for-android/how-to/define-api-in-typescript). Alternatively, you can use Portals' [Web Vitals plugin](https://ionic.io/docs/portals/for-web/web-vitals) to supply a callback that will display the Portal after First Contentful Paint (FCP) has completed.
+
+## Optimize the web app's load time
```
Comment (kensodemann) [inline docs/portals/knowledge-base/addressing-portal-load-times.md:19]: This seems the most important, and I would likely put it first.

File: docs/portals/knowledge-base/addressing-portal-load-times.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+## Mask with a loading view
+
+Use a loading view to prevent users from seeing "blank Portals". Add the Portal component to the view stack with zero opacity to hide it as it loads and display the loading view until the web app is ready to be presented. Once ready, remove the loading view and set the Portal component's opacity to 1 to display the rendered content.
+
+There are two ways to approach signaling when the Portal is ready to be made visible. You can use the [communication mechanisms](https://ionic.io/docs/portals/choosing-a-communication) built into Portals to let the web app notify the native code layer when it is ready for the Portal to be visible. Tutorials on implementing this approach can be found [here for iOS](https://ionic.io/docs/portals/for-ios/how-to/define-api-in-typescript) and [here for Android](https://ionic.io/docs/portals/for-android/how-to/define-api-in-typescript). Alternatively, you can use Portals' [Web Vitals plugin](https://ionic.io/docs/portals/for-web/web-vitals) to supply a callback that will display the Portal after First Contentful Paint (FCP) has completed.
```
Comment (kensodemann) [inline docs/portals/knowledge-base/addressing-portal-load-times.md:17]: I think it reads better if you avoid "here" in links. For example:

"Please see our tutorials that implement this approach for iOS and Android."

Then the links are just on "iOS" and "Android" and it reads more naturally.

**Review by kensodemann — COMMENTED** [review]
Just some organizational suggestions, but I am fine with it as-is too.
